src/buildApplication/generateFunctionsMap.ts

```diff
@@ -53,6 +56,51 @@ export async function generateFunctionsMap(
 	return processingResults;
 }
 
+/**
+ * Process the invalid functions and check whether and valid function was created in the functions
+ * map to override it.
+ *
+ * The build output sometimes generates invalid functions at the root, while still creating the
+ * valid functions. With the base path and route groups, it might create the valid edge function
+ * inside a folder for the route group, but create an invalid one that maps to the same path
+ * at the root.
+ *
+ * When we process the directory, we might add the valid function to the map before we process the
+ * invalid one, so we need to check if the invalid one was added to the map and remove it from the
+ * set if it was.
+ *
+ * If the invalid function is an RSC function (e.g. `path.rsc`) and doesn't have a valid squashed
+ * version, we check if a squashed non-RSC function exists (e.g. `path`) and use this instead. RSC
+ * functions are the same as non-RSC functions, per the Vercel source code.
+ * https://github.com/vercel/vercel/blob/main/packages/next/src/server-build.ts#L1193
+ *
+ * @param processingResults Object containing the results of processing the current function directory.
+ */
+async function tryToFixInvalidFunctions({
+	functionsMap,
+	invalidFunctions,
+}: DirectoryProcessingResults): Promise<void> {
+	if (invalidFunctions.size > 0) {
+		for (const rawPath of invalidFunctions) {
+			const formattedPath = formatRoutePath(rawPath);
+
+			if (
+				functionsMap.has(formattedPath) ||
+				functionsMap.has(stripIndexRoute(formattedPath))
+			) {
+				invalidFunctions.delete(rawPath);
+			} else if (formattedPath.endsWith('.rsc')) {
+				const value = functionsMap.get(formattedPath.replace(/\.rsc$/, ''));
+
+				if (value) {
+					functionsMap.set(formattedPath, value);
+					invalidFunctions.delete(rawPath);
+				}
+			}
+		}
+	}
```

Comment:
if instead of checking invalidFunctions.size > 0, you could check if it's equal 0 and return, so that we don't need to nest everything here, it might help in readability

not a big deal, what do you think? do you have a preference?

```suggestion
	if (invalidFunctions.size === 0) {
		return;
	}

	for (const rawPath of invalidFunctions) {
		const formattedPath = formatRoutePath(rawPath);

		if (
			functionsMap.has(formattedPath) ||
			functionsMap.has(stripIndexRoute(formattedPath))
		) {
			invalidFunctions.delete(rawPath);
		} else if (formattedPath.endsWith('.rsc')) {
			const value = functionsMap.get(formattedPath.replace(/\.rsc$/, ''));

			if (value) {
				functionsMap.set(formattedPath, value);
				invalidFunctions.delete(rawPath);
			}
		}
	}
```